### DISCUSSION TITLE
Has anyone else, Using Integrations such as Palo successfully patched to 2.4.20?

### DISCUSSION THREAD

**Discussion by Phillip-Samson** [Security-Onion-Solutions/securityonion#11514] [2.4]
### Version

2.4.20

### Installation Method

Network installation on Red Hat derivative like Oracle, Rocky, Alma, etc.

### Description

upgrading

### Installation Type

Distributed

### Location

cloud

### Hardware Specs

Exceeds minimum requirements

### CPU

16

### RAM

64gig

### Storage for /

500gig

### Storage for /nsm

8tb

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Since patching from 2.4.10 hotfix to 2.4.20 all now Darktrace and PaloAlto indexing is no longer working. Is there a new way of adding integrations. Nothing seems to be working 100% and I have now resigned myself to rolling back
I do use a Forward Node for receiving the Palo Logs but like I said it is not working
the palo DataStream appears as logs-panw.panos-default is this normal?
Do I need to uninstall all integrations then do the update and reinstall.
Several elastic_agents are now not functioning normally and I can seem stop find anything in the logs to explain why.
I am lost and frustrated
Is anyone else experiencing these issues or am I hosed and need to rebuild. 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by weslambert** [comment]
I would try the following:

- Uninstall any integrations you previously configured manually
- Uninstall the assets for the integrations, within the Settings pane
- Run so-elastic-fleet-package-upgrade
- Run so-elasticsearch-templates-load
- Enable the integration from the UI

**Reply by Phillip-Samson** [reply]
Thanks @weslambert I will give it a try

**Reply by defensivedepth** [reply]
There is also a known issue with Elastic Defend integration that we are working on a fix for.

**Reply by weslambert** [reply]
I would also remove any current indices related to integrations if they exist to avoid confusion, since they might not have the pipeline setting applied.

**Comment by 8bitjoe** [comment]
What index Template is the data stream selecting?

**Reply by protecme** [reply]
Hello

after testing this : 
Uninstall any integrations 
Uninstall the assets for the integrations, within the Settings pane
Run so-elastic-fleet-package-upgrade
Run so-elasticsearch-templates-load
Enable the integration from the UI ( PANW and netflow ) 

it doesn't work.
And now a have a problem on my sonde : 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/cfd1d77e-9a6c-431a-868d-f752c93d0585)

do you have an command to correct the problem 
for Docker image generation or remote reinstallation  ?

thanks

**Comment by protecme** [comment]
OK after reboot manager and sensor.
but PANW and netflow not working.

**Comment by weslambert** [comment]
Okay, let's take a step back. 

Are data streams for the integrations being created, and are events showing up, but not parsed?

What do you mean when you say that some Elastic agents are not functioning normally? Are there errors in the Elastic Fleet page?

**Comment by protecme** [comment]
hi,
thanks or the answer.
Data streams for integrations are created, and events appear, but are not analyzed?

I followed the integration of the netflow module as in my current post here: Elasticagent integration netflow #11487

the data doesn't seem to arrive in the ELK part the netflow dashboard is empty
I've tried playing with id-patterns in dashboard filters to place either one or the other
logs-netflow.log
so-logs-netflow.log

but it remains empty

What do you mean when you say that some Elastic agents don't work normally? Are there any errors on the Elastic fleet page?

At Elastic fleet level. no errors
I tried creating a dedicated policy for the probe, but it had the same effect as placing it in the global
as far as receiving netflow on the forwarder , I'm sure it's working, because in the iptable  rules match and as far as packets received from the switch are concerned, this seems correct and the tcpdump confirms that packets have been received.

despite this, I can't find any trace of netflow information on the SO dashboard or in Kibana.

if I look at the agent logs via the fleet server, I can't find anything related to the netflow module and no errors, only information on the agent's state of health.

the configuration remains as follows: 
-Fresh install so 2.4.10 hotfixe
-Upgrade to 2.4.20
-Install integration module netflow on fleet server

nothing else

unfortunately I can't find any data in SO or Kibana

what do you need ( Logs / screen ) to analyze the problem ?

many thanks

**Reply by weslambert** [reply]
The netflow integration works for me on a fresh installation of Security Onion 2.4.20.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/16829864/4190d3dc-c9d4-4150-8cf7-5dd5c073e4c7)

I will need to test an upgrade to see if it changes anything.

Are you able reach port 2005 or 9001 using something like nc from another machine?

nc -vzu $ip 2055

**Reply by Phillip-Samson** [reply]
We are not starting fresh, this is an update
I deleted the old integrations and all artifacts that belonged to them 
run an update 
and now.........
I can't login to the SOC 


I tried creating a new user sill getting the same error bad username and password

**Comment by weslambert** [comment]
> The netflow integration works for me on a fresh installation of Security Onion 2.4.20.
> 
> ![image](https://user-images.githubusercontent.com/16829864/274679944-4190d3dc-c9d4-4150-8cf7-5dd5c073e4c7.png)
> 
> I will need to test an upgrade to see if it changes anything.
> 
> Are you able reach port 2005 or 9001 using something like nc from another machine?
> 
> nc -vzu $ip 2055

I just tested an upgrade from 2.4.10 with the netflow integration configured and ingesting and parsing data to 2.4.20 with no regression.

**Comment by Phillip-Samson** [comment]
OK! Now I have some info to work with.
First, I rolled back, and then I uninstalled all of the integration indexes templates and components that I was seeing issues with 
Okta, Palo, Darktrace, SentinelOne
I then patched to 20 and couldn't log in, so I changed the SO version number back to 10 and reran soup. 
Confirmed it patched, and I was able to log in
I then ran your solution
Run so-elastic-fleet-package-upgrade - successful!
Run so-elasticsearch-templates-load - errors
logs-panw.panos
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [logs-panw.panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [logs-panw.panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
logs-sentinel_one.activity-so
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [logs-sentinel_one.activity-so] has index patterns [logs-sentinel_one.activity-*] matching patterns from existing templates [logs-sentinel_one.activity] with patterns (logs-sentinel_one.activity => [logs-sentinel_one.activity-*]) that have the same priority [200], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [logs-sentinel_one.activity-so] has index patterns [logs-sentinel_one.activity-*] matching patterns from existing templates [logs-sentinel_one.activity] with patterns (logs-sentinel_one.activity => [logs-sentinel_one.activity-*]) that have the same priority [200], multiple index templates may not match during index creation, please use a different priority"},"status":400}
logs-sentinel_one-alert-so
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [logs-sentinel_one-alert-so] has index patterns [logs-sentinel_one-alert-*] matching patterns from existing templates [logs] with patterns (logs => [logs-*-*]) that have the same priority [100], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [logs-sentinel_one-alert-so] has index patterns [logs-sentinel_one-alert-*] matching patterns from existing templates [logs] with patterns (logs => [logs-*-*]) that have the same priority [100], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-darktrace.ai_analyst_alert
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-darktrace.ai_analyst_alert] has index patterns [logs-darktrace.ai_analyst_alert-*] matching patterns from existing templates [so-darktrace-ai_analyst_alert] with patterns (so-darktrace-ai_analyst_alert => [logs-darktrace.ai_analyst_alert-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-darktrace.ai_analyst_alert] has index patterns [logs-darktrace.ai_analyst_alert-*] matching patterns from existing templates [so-darktrace-ai_analyst_alert] with patterns (so-darktrace-ai_analyst_alert => [logs-darktrace.ai_analyst_alert-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-darktrace-model_breach_alert
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-darktrace-model_breach_alert] has index patterns [logs-sentinel_one-alert-*] matching patterns from existing templates [logs] with patterns (logs => [logs-*-*]) that have the same priority [100], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-darktrace-model_breach_alert] has index patterns [logs-sentinel_one-alert-*] matching patterns from existing templates [logs] with patterns (logs => [logs-*-*]) that have the same priority [100], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-logs-darktrace.ai_analyst_alert
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-darktrace.ai_analyst_alert] has index patterns [logs-darktrace.ai_analyst_alert-*] matching patterns from existing templates [so-darktrace-ai_analyst_alert] with patterns (so-darktrace-ai_analyst_alert => [logs-darktrace.ai_analyst_alert-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-darktrace.ai_analyst_alert] has index patterns [logs-darktrace.ai_analyst_alert-*] matching patterns from existing templates [so-darktrace-ai_analyst_alert] with patterns (so-darktrace-ai_analyst_alert => [logs-darktrace.ai_analyst_alert-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-logs-o365.audit
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-o365.audit] has index patterns [logs-o365.audit-*] matching patterns from existing templates [logs-o365.audit] with patterns (logs-o365.audit => [logs-o365.audit-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-o365.audit] has index patterns [logs-o365.audit-*] matching patterns from existing templates [logs-o365.audit] with patterns (logs-o365.audit => [logs-o365.audit-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-logs-okta.system
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-okta.system] has index patterns [logs-okta.system-*] matching patterns from existing templates [logs-okta.system] with patterns (logs-okta.system => [logs-okta.system-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-okta.system] has index patterns [logs-okta.system-*] matching patterns from existing templates [logs-okta.system] with patterns (logs-okta.system => [logs-okta.system-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-logs-panw.panos-default
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-panw.panos-default] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-panw.panos-default] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-logs-panw-panos
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-panw-panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-panw-panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}
so-logs-panw.panos
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-panw.panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-panw.panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}


So it is just as I thought. somewhere, the integration info is being held onto, and I can't seem to purge it.

**Reply by weslambert** [reply]
If you manually configured the integration before updating, try checking /opt/so/conf/elasticsearch/templates/index to see if the old template is there. If it is, remove it.

**Reply by weslambert** [reply]
Keep in mind, it will get added back if you still have the configuration in SOC. I would remove and copy that into a backup file just in case you need it.

**Reply by Phillip-Samson** [reply]
confirmed **
-rw-r--r--. 1 root          root     415 Oct  4 20:47 logs-darktrace.ai_analyst_alert-template.json
-rw-r--r--. 1 root          root     421 Oct  4 21:20 logs-darktrace.model_breach_alert-template.json
-rw-r--r--. 1 root          root     367 Aug 28 18:42 logs-iis.access-template.json
-rw-r--r--. 1 root          root     364 Aug 28 18:57 logs-iis.error-template.json
-rw-r--r--. 1 root          root     367 Aug 22 19:29 logs-o365.audit-template.json
-rw-r--r--. 1 root          root     370 Aug 28 17:24 logs-okta.system-template.json
-rw-r--r--. 1 root          root     367 Oct  8 14:14 logs-panw.panos-default-template.json
-rw-r--r--. 1 root          root     367 Oct  7 15:47 logs-panw-panos-template.json
-rw-r--r--. 1 root          root     367 Oct 11 14:07 logs-panw.panos-template.json
-rw-r--r--. 1 root          root     400 Sep 21 02:54 logs-sentinel_one.activity-so-template.json
-rw-r--r--. 1 root          root     391 Sep 21 11:21 logs-sentinel_one-alert-so-template.json
-rw-r--r--. 1 root          root     391 Sep 21 03:00 logs-sentinel_one.alert-so-template.json
-rw-r--r--. 1 root          root     394 Sep 21 03:02 logs-sentinel_one.threat-so-template.json
-rw-r--r--. 1 root          root     394 Aug 24 00:17 logs-sentinel_one.threat-template.json

[root@orseconip01 index]# rm -rf *darktrace.*
[root@orseconip01 index]# rm -rf *sentinel_one*
[root@orseconip01 index]# rm -rf *okta*
[root@orseconip01 index]# rm -rf *panw*
[root@orseconip01 index]# rm -rf *o365* 
[root@orseconip01 index]# rm -rf *iis*



So that is super interesting. If you use the advanced template add, it doesn't place the info in local it leaves it in conf and never gets rewritten?


after running the above command, I still receive the following  error while loading th templates

so-logs-darktrace.ai_analyst_alert
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-darktrace.ai_analyst_alert] has index patterns [logs-darktrace.ai_analyst_alert-*] matching patterns from existing templates [so-darktrace-ai_analyst_alert] with patterns (so-darktrace-ai_analyst_alert => [logs-darktrace.ai_analyst_alert-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-darktrace.ai_analyst_alert] has index patterns [logs-darktrace.ai_analyst_alert-*] matching patterns from existing templates [so-darktrace-ai_analyst_alert] with patterns (so-darktrace-ai_analyst_alert => [logs-darktrace.ai_analyst_alert-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}

so-logs-panw.panos
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"index template [so-logs-panw.panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"}],"type":"illegal_argument_exception","reason":"index template [so-logs-panw.panos] has index patterns [logs-panw.panos-*] matching patterns from existing templates [logs-panw-panos] with patterns (logs-panw-panos => [logs-panw.panos-*]) that have the same priority [501], multiple index templates may not match during index creation, please use a different priority"},"status":400}

**Reply by weslambert** [reply]
You might also need to delete the older index templates from Elasticsearch via Kibana. It looks like they are conflicting.

**Comment by protecme** [comment]
Hello, 

I have just redone an installation in version 2.4.20
the hotfix is not yet deployed

here's how I configured SO for the netflow part

Firewall : 
Source of netflow in : Firewall-->hostgroups-->customhostgroup0 : 
<img width="1162" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/fb9dfbe1-b089-41ce-a1eb-c2af8947c29e">
ONLY for my sensor node

Destination port : Firewall -->portgroups-->customportgroup0 UDP and TCP
<img width="1163" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/fc8be321-c7d5-4ccf-9be7-29f1bbaec8b5">
ONLY dor my sensor node 

Activation on Sensor node : Firewall -->role-->sensor-->chain-->input-->hostsgroup-->customhostgroup0==>portgroups

<img width="1158" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/d194a698-42ce-4c70-a17c-eba865c3692c">
ONLY for my sensor node
After 15mn it's ok on my node : 
<img width="723" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/eb9b2018-0cdb-4c65-a99a-775dff0d456d">


On fleet server : 
<img width="1231" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/739c42fd-1ebe-4599-b6b1-3ece570395b8">
<img width="1004" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/67becdd1-7d0e-49e8-b5c8-a748d61ba60b">
<img width="1290" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/99ffd64f-a3ae-4c46-b771-4bcd2e96c9fe">


but if I test a connection from the manager 
10.x.x.1 to the probe in 10.x.x.3 
<img width="346" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/a2cedf87-b387-465e-a68c-d6056dd0e3d2">

but the rule match....
and no result to dashboard SO or Kibana : 
<img width="1386" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/6f1c4add-71e5-4ed6-93d5-22b92af5d8e3">

<img width="571" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/d6f1e61d-fc9e-4ceb-8f1f-f055bf341a56">

<img width="1172" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/37100653/f39cb83d-cb6b-46ce-a1c2-a181e4b756b7">
can you tell me how you configured the firewall part, as I'm not sure how to configure it?

Many thanks

**Reply by Phillip-Samson** [reply]
so for me I had to roll back to a backup of the entire servers both Manager-search  and forward node 
I confirmed everything was working then i deleted all  indexes templates components and removed all integrations that I customized\
ran the patch and then reinstalled all of the integrations 

everything was working except Palo and Cisco integrations. 
I run a tcp[dump -XX -I any port 9003 for FTD and 9001 for palo and I seee the data coming into the forward node 
I have seen short bursts be sent to the index. One short burst of about 300 logs sent to both indexes and then nothing
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/8fbb466e-ecbb-4f75-9acd-9c1379872c2d)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/fd9d0690-689e-4908-92cb-bf550ef9d11c)

**Reply by Phillip-Samson** [reply]
I think this worked
I have tried leaving it as localhost didn't work
I then had it set as 0.0.0.0, and I would see bursts of data for a moment, then nothing
yesterday at 5, I followed your advice and added the IP of the forward node, and now I have millions of logs
